internal/pkg/mapper/mapper.go

```diff
@@ -0,0 +1,255 @@
+		isMatch, err := isMatchPredicate(rule, node, typeURL)
+		if err != nil {
+			return false, err
+		}
+		if !isMatch {
+			return false, nil
+		}
+	}
+	return true, nil
+}
+
+func isOrMatch(matchPredicate *matchPredicate, node core.Node, typeURL string) (bool, error) {
+	matchset := matchPredicate.GetOrMatch()
+	for _, rule := range matchset.GetRules() {
+		isMatch, err := isMatchPredicate(rule, node, typeURL)
+		if err != nil {
+			return false, err
+		}
+		if isMatch {
+			return true, nil
+		}
+	}
+	return false, nil
+}
+
+func isNotMatch(matchPredicate *matchPredicate, node core.Node, typeURL string) (bool, error) {
+	predicate := matchPredicate.GetNotMatch()
+	isMatch, err := isMatchPredicate(predicate, node, typeURL)
+	if err != nil {
+		return false, err
+	}
+	return !isMatch, nil
+}
+
+func isRequestTypeMatch(matchPredicate *matchPredicate, typeURL string) bool {
+	predicate := matchPredicate.GetRequestTypeMatch()
+	for _, t := range predicate.GetTypes() {
+		if t == typeURL {
+			return true
+		}
+	}
+	return false
+}
+
+func isNodeTypeMatch(matchPredicate *matchPredicate, node core.Node) (bool, error) {
+	predicate := matchPredicate.GetRequestNodeMatch()
+	switch predicate.GetField() {
+	case aggregationv1.NodeFieldType_NODE_CLUSTER:
+		return compare(predicate, node.GetCluster())
+	case aggregationv1.NodeFieldType_NODE_ID:
+		return compare(predicate, node.GetId())
+	case aggregationv1.NodeFieldType_NODE_LOCALITY_REGION:
+		return compare(predicate, node.GetLocality().GetRegion())
+	case aggregationv1.NodeFieldType_NODE_LOCALITY_ZONE:
+		return compare(predicate, node.GetLocality().GetZone())
+	case aggregationv1.NodeFieldType_NODE_LOCALITY_SUBZONE:
+		return compare(predicate, node.GetLocality().GetSubZone())
+	default:
+		return false, nil
+	}
+}
+
+func compare(requestNodeMatch *aggregationv1.MatchPredicate_RequestNodeMatch, nodeValue string) (bool, error) {
+	exactMatch := requestNodeMatch.GetExactMatch()
+	if exactMatch != "" {
+		return nodeValue == exactMatch, nil
+	}
+
+	regexMatch := requestNodeMatch.GetRegexMatch()
+	if regexMatch != "" {
+		match, err := regexp.MatchString(regexMatch, nodeValue)
+		if err != nil {
+			return false, err
+		}
+		return match, nil
+	}
+
+	return false, nil
+}
+
+func isMatchPredicate(matchPredicate *matchPredicate, node core.Node, typeURL string) (bool, error) {
+	if matchPredicate.GetAnyMatch() {
+		return isAnyMatch(matchPredicate), nil
+	} else if matchPredicate.GetAndMatch() != nil {
+		return isAndMatch(matchPredicate, node, typeURL)
+	} else if matchPredicate.GetOrMatch() != nil {
+		return isOrMatch(matchPredicate, node, typeURL)
+	} else if matchPredicate.GetNotMatch() != nil {
+		return isNotMatch(matchPredicate, node, typeURL)
+	} else if matchPredicate.GetRequestTypeMatch() != nil {
+		return isRequestTypeMatch(matchPredicate, typeURL), nil
+	} else if matchPredicate.GetRequestNodeMatch() != nil {
+		return isNodeTypeMatch(matchPredicate, node)
+	}
+
+	return false, nil
```

Comment:
Thoughts on error-ing here for an unsupported predicate? Or warn logging?

Reply:
yes...there's no logging now. Will add logs today

Reply:
On an unsupported predicate qs, we should also be doing `Validate()` to check consistency of the proto.